pkg/topology/kademlia/kademlia.go

```diff
@@ -231,43 +231,35 @@ func (k *Kad) connectBalanced(wg *sync.WaitGroup, peerConnChan chan<- *peerConnI
 		}
 
 		binPeers := k.knownPeers.BinPeers(uint8(i))
+		binConnectedPeers := k.connectedPeers.BinPeers(uint8(i))
 
 		for j := range k.commonBinPrefixes[i] {
 			pseudoAddr := k.commonBinPrefixes[i][j]
 
-			closestConnectedPeer, err := closestPeer(k.connectedPeers, pseudoAddr, noopSanctionedPeerFn)
+			// Connect to closest known peer which we haven't tried connecting to recently.
+
+			_, err := nClosePeerInSlice(binConnectedPeers, pseudoAddr, noopSanctionedPeerFn, uint8(i+k.bitSuffixLength+1))
 			if err != nil {
-				if errors.Is(err, topology.ErrNotFound) {
-					break
+				if !errors.Is(err, topology.ErrNotFound) {
+					return
 				}
-				k.logger.Errorf("closest connected peer: %v", err)
-				continue
-			}
-
-			closestConnectedPO := swarm.ExtendedProximity(closestConnectedPeer.Bytes(), pseudoAddr.Bytes())
-			if int(closestConnectedPO) >= i+k.bitSuffixLength+1 {
+			} else {
 				continue
 			}
 
-			// Connect to closest known peer which we haven't tried connecting to recently.
-			closestKnownPeer, err := closestPeerInSlice(binPeers, pseudoAddr, skipPeers)
+			closestKnownPeer, err := nClosePeerInSlice(binPeers, pseudoAddr, skipPeers, uint8(i+k.bitSuffixLength+1))
 			if err != nil {
 				if errors.Is(err, topology.ErrNotFound) {
-					break
+					continue
+				} else {
+					return
```

Comment:
return here?

Reply:
yes, no other error than topology.ErrNotFound is expected here

Reply:
this return is also non-consequential, it can be removed

Reply:
why return and not `break`? please see the comment on the other change